## Notebook to verify Catalog Demand Project answers/ equation

In [2]:
import pandas as pd


In [3]:
df= pd.read_excel('Step 3 - Expected Profit Calculation.xlsx', header=1)
df.head()

Name              Customer_Segment  Customer_ID  \
0       A Giametti             Loyalty Club Only       2213.0   
1     Abby Pierson  Loyalty Club and Credit Card       2785.0   
2    Adele Hallman             Loyalty Club Only       2931.0   
3  Alejandra Baird             Loyalty Club Only       2231.0   
4     Alice Dewitt             Loyalty Club Only       2530.0   

               Address        City State      ZIP  Store_Number  \
0    5326 S Lisbon Way  Centennial    CO  80015.0         105.0   
1    4344 W Roanoke Pl      Denver    CO  80236.0         101.0   
2   5219 S Delaware St   Englewood    CO  80110.0         101.0   
3     2301 Lawrence St      Denver    CO  80205.0         103.0   
4  5549 S Hannibal Way  Centennial    CO  80015.0         104.0   

   Avg_Num_Products_Purchased  #_Years_as_Customer      ...       \
0                         3.0                  0.2      ...        
1                         6.0                  0.6      ...        
2                         7.0                  0.9      ...        
3                         2.0                  0.6      ...        
4                         4.0                  0.5      ...        

   Loyalt Club Only  Store Mailing List  Credit Card Only  Average Sales  \
0            149.36                 NaN               NaN         355.04   
1               NaN                 NaN               NaN         987.18   
2            149.36                 NaN               NaN         622.96   
3            149.36                 NaN               NaN         288.06   
4            149.36                 NaN               NaN         422.02   

   Score_Yes.1  Expected Sale  Sales Margin  Ending Sales Value  \
0     0.305036     108.299913           0.5           54.149956   
1     0.472725     466.664208           0.5          233.332104   
2     0.578882     360.620237           0.5          180.310119   
3     0.305138      87.897998           0.5           43.948999   
4     0.387706     163.619625           0.5           81.809812   

   Cost of Sending Catalogs  Final Profit  
0                       6.5     47.649956  
1                       6.5    226.832104  
2                       6.5    173.810119  
3                       6.5     37.448999  
4                       6.5     75.309812  

[5 rows x 25 columns]

In [4]:
df.count()

Name                            250
Customer_Segment                250
Customer_ID                     250
Address                         250
City                            250
State                           250
ZIP                             250
Store_Number                    250
Avg_Num_Products_Purchased      250
#_Years_as_Customer             250
Score_No                        250
Score_Yes                       250
Y Intercept                     250
Avg_Num_Products_Purchased.1    250
Loyalty Club and Credit Card     26
Loyalt Club Only                122
Store Mailing List               20
Credit Card Only                 82
Average Sales                   250
Score_Yes.1                     250
Expected Sale                   250
Sales Margin                    250
Ending Sales Value              250
Cost of Sending Catalogs        250
Final Profit                    251
dtype: int64

In [5]:
df = df.fillna(0)

In [6]:
df.tail()

Name   Customer_Segment  Customer_ID             Address  \
246  Wendy Crenshaw  Loyalty Club Only        304.0     12483 W 8th Ave   
247    Wendy Renaud  Loyalty Club Only       2629.0     700 S Alton Way   
248      Wilma Reid   Credit Card Only       1917.0     6763 W Yale Ave   
249     Yvette Kent   Credit Card Only        762.0  1071 Worchester St   
250               0                  0          0.0                   0   

         City State      ZIP  Store_Number  Avg_Num_Products_Purchased  \
246    Golden    CO  80401.0         106.0                         3.0   
247    Denver    CO  80247.0         100.0                         6.0   
248  Lakewood    CO  80227.0         101.0                         7.0   
249    Aurora    CO  80011.0         100.0                         5.0   
250         0     0      0.0           0.0                         0.0   

     #_Years_as_Customer      ...       Loyalt Club Only  Store Mailing List  \
246                  0.6      ...                 149.36                 0.0   
247                  0.1      ...                 149.36                 0.0   
248                  0.2      ...                   0.00                 0.0   
249                  0.5      ...                   0.00                 0.0   
250                  0.0      ...                   0.00                 0.0   

     Credit Card Only  Average Sales  Score_Yes.1  Expected Sale  \
246               0.0         355.04     0.192800      68.451749   
247               0.0         555.98     0.423456     235.432947   
248               0.0         772.32     0.259251     200.224358   
249               0.0         638.36     0.203650     130.002030   
250               0.0           0.00     0.000000       0.000000   

     Sales Margin  Ending Sales Value  Cost of Sending Catalogs  Final Profit  
246           0.5           34.225874                       6.5     27.725874  
247           0.5          117.716474                       6.5    111.216474  
248           0.5          100.112179                       6.5     93.612179  
249           0.5           65.001015                       6.5     58.501015  
250           0.0            0.000000                       0.0  21987.957028  

[5 rows x 25 columns]

#### Replace by the Final Equation to be Checked. 
<p>Answer is actually: 
<p>AverageSaleAmount = 303.46 - 149.36 * Loyalty Club Only + 281.84 * Loyalty Club and Credit Card -245.42 *
Store Mailing List + 66.98 * Avg_Num_Products_Purchased

In [7]:
df['FinalSale']= 21776.6709 + 67.09*df['Avg_Num_Products_Purchased'] -150.2038 * df['Loyalt Club Only'] + 282.8251* df['Loyalty Club and Credit Card'] -245.2238 * df['Store Mailing List']

In [8]:
df['FinalSale'].head()

0      -456.498668
1    101890.637084
2      -188.138668
3      -523.588668
4      -389.408668
Name: FinalSale, dtype: float64

In [9]:
df.sum()['FinalSale']

3676144.469467998